### PR TITLE
Fix billing email being parsed as a URL link when beginning of email …

### LINKED ISSUE #25099: Order email bounce back if the customer email address starts with www.
On receiving an order from a customer that has an email address starting with "www." such as "www.joebloggs@gmail.com", woocommerce or wordpress automatically turns it into a web address, adding http:// in front of it and so all the emails sent from woocommerce including order status emails bounce back.

### PR DESCRIPTION
**Pull request by roykho**
…contains www closes #25099

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Make email clickable without it being transformed into a URL link because the email contains `www` at the beginning.

Closes #25099 

### How to test the changes in this Pull Request:

1. Make a test order with a billing email address that starts with `www` like `www.johndoe@test.com`
2. Go to the order detail page in admin and check the billing email and ensure the email link do not have `http` added to it.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Fix - Email address starting with `www` being displayed as a URL link in the admin order details page.

### CODE REVIEW

File: includes/admin/meta-boxes/class-wc-meta-box-order-data.php [hunk 1]
```diff
@@ -344,6 +344,8 @@ public static function output( $post ) {
 
 								if ( 'billing_phone' === $field_name ) {
 									$field_value = wc_make_phone_clickable( $field_value );
+								} else if ( 'billing_email' === $field_name ) {
+									$field_value = '<a href="mailto:' . $field_value . '">' . $field_value . '</a>';
```
Comment (claudiosanches) [inline includes/admin/meta-boxes/class-wc-meta-box-order-data.php:348]:
```suggestion
									$field_value = '<a href="' . esc_url( 'mailto:' . $field_value ) . '">' . $field_value . '</a>';
```

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Good work, just missing escape the URL.

File: includes/admin/meta-boxes/class-wc-meta-box-order-data.php (hunk 1 above)
Comment (roykho) [reply includes/admin/meta-boxes/class-wc-meta-box-order-data.php:348]: WTH! I did escape...not sure why this got pushed up!! Also the $field_value should be escaped too..

Comment (roykho) [reply includes/admin/meta-boxes/class-wc-meta-box-order-data.php]: Fixed...another round plz.

**Review by claudiosanches — APPROVED** [review]
Looks great!

**Pull request merged**